modules/account-lib/src/coin/baseCoin/ed25519KeyPair.ts

```diff
@@ -1,22 +1,98 @@
+import * as nacl from 'tweetnacl';
+import * as hex from '@stablelib/hex';
+import * as stellar from 'stellar-sdk';
+import { Seed, generateAccountFromSeed, generateAccount } from 'algosdk';
 import { BaseKeyPair } from './baseKeyPair';
 import { AddressFormat } from './enum';
 import { NotImplementedError } from './errors';
+import { isPrivateKey, isPublicKey, isSeed, DefaultKeys, KeyPairOptions, ByteKeys } from './iface';
+
+const DEFAULT_SEED_SIZE_BYTES = 32;
 
 export abstract class Ed25519KeyPair implements BaseKeyPair {
+  protected keyPair: ByteKeys;
+  protected source?: KeyPairOptions;
+
+  /**
+   * Public constructor. By default, creates a key pair with a random master seed.
+   *
+   * @param {KeyPairOptions} source Either a master seed, a private key (extended or raw), or a public key
+   */
+  protected constructor(source?: KeyPairOptions) {
+    let naclKeyPair;
+    if (!source) {
+      const seed = nacl.randomBytes(DEFAULT_SEED_SIZE_BYTES);
+      naclKeyPair = nacl.sign.keyPair.fromSeed(seed);
+      this.setKeyPair(naclKeyPair);
+    } else if (isSeed(source)) {
+      naclKeyPair = nacl.sign.keyPair.fromSeed(new Uint8Array(source.seed));
+      this.setKeyPair(naclKeyPair);
+    } else if (isPrivateKey(source)) {
+      this.recordKeysFromPrivateKey(source.prv);
+    } else if (isPublicKey(source)) {
+      this.recordKeysFromPublicKey(source.pub);
+    } else {
+      throw new Error('Invalid key pair options');
+    }
+  }
+
+  private setKeyPair(naclKeyPair: nacl.SignKeyPair): void {
+    this.keyPair = { prv: naclKeyPair.secretKey, pub: naclKeyPair.publicKey };
+  }
+
   getAddress(format?: AddressFormat): string {
-    throw new NotImplementedError('getAddress not implemented');
+    throw new NotImplementedError('It is not possible to derive an address from an ed25519 keypair');
   }
 
-  getKeys(): any {
-    // TODO: implement from stellar seed, etc.
-    throw new NotImplementedError('getKeys not implemented');
+  getKeys(): ByteKeys {
+    const result: ByteKeys = {
+      pub: this.keyPair.pub,
+    };
+
+    if (this.keyPair.prv) {
+      result.prv = this.keyPair.prv;
+    }
+
+    return result;
   }
 
   recordKeysFromPrivateKey(prv: string): void {
-    throw new NotImplementedError('recordKeysFromPrivateKey not implemented');
+    const decodedPrv = hex.decode(prv);
+    const naclKeyPair = nacl.sign.keyPair.fromSecretKey(decodedPrv);
+    this.setKeyPair(naclKeyPair);
   }
 
   recordKeysFromPublicKey(pub: string): void {
-    throw new NotImplementedError('recordKeysFromPublicKey not implemented');
+    const decodedPub = hex.decode(pub);
```

Comment:
raw public keys can be different than the protocol specific keys. This method should allow parsing either raw or protocol-specific formats.

Reply:
Great, I've created a new PR for keypair: https://github.com/BitGo/BitGoJS/pull/807
And I am going to be working on these comments there.